### ISSUE TITLE
macOS High Sierra 10.13 (17A365) - gPodder does not launch

### ISSUE THREAD

**Issue by jonathangodoy** [gpodder/gpodder#347]
Hello,

gPodder 3.9.3 does not launch on macOS High Sierra, 10.13 (17A365).

I thought it was an issue with the beta version that I was running, but it still doesn't launch on the final version of macOS High Sierra. It bounces a few times in the dock and closes right away.

It's still working on OS X El Capitan (10.11.6).

Hope to see an update for this.

Thanks!

**Comment by elelay** [comment]
Hello,
I don't have a High Sierra capable device right now, so I'll rely on you for debugging ;-)
Can you launch the Application/Utilities/Console application and clear display before launching gPodder?
Then report back what log messages appear to be related when you launch gPodder.
Another way would be to run gPodder from the Terminal: type `/Applications/gPodder.app/Contents/MacOS/gpodder` and report what log you see.
Thanks,

**Comment by jonathangodoy** [comment]
Hi,

Thanks for looking into this.

When I launch gPodder from terminal, I get this:

```
computer:~ user$ /Applications/gPodder.app/Contents/MacOS/gpodder
(Re)generating /Users/user/Library/Application Support/gPodder/openssl/cert.pem
I: make_cert_pem.py /Applications/gPodder.app/Contents/Resources/bin/openssl /Users/user/Library/Application Support/gPodder/openssl/cert.extracted.pem
I: extracted 164 certificates
I: of which 163 are valid certificates
I: updated /Users/user/Library/Application Support/gPodder/openssl/cert.extracted.pem with 163 certificates
1506561974.927534 [gpodder.log] ERROR: Uncaught exception: Traceback (most recent call last):
  File "/Applications/gPodder.app/Contents/Resources/bin/gpodder", line 155, in <module>
    main()
  File "/Applications/gPodder.app/Contents/Resources/bin/gpodder", line 148, in main
    from gpodder.gtkui import main
  File "/Applications/gPodder.app/Contents/Resources/lib/python2.7/site-packages/gpodder/gtkui/main.py", line 22, in <module>
    import gtk
  File "/Applications/gPodder.app/Contents/Resources/lib/python2.7/site-packages/gtk-2.0/gtk/__init__.py", line 30, in <module>
    import gobject as _gobject
  File "/Applications/gPodder.app/Contents/Resources/lib/python2.7/site-packages/gobject/__init__.py", line 26, in <module>
    from glib import spawn_async, idle_add, timeout_add, timeout_add_seconds, \
  File "/Applications/gPodder.app/Contents/Resources/lib/python2.7/site-packages/glib/__init__.py", line 22, in <module>
    from glib._glib import *
ImportError: dlopen(/Applications/gPodder.app/Contents/Resources/lib/python2.7/site-packages/glib/_glib.so, 2): Symbol not found: _inflateValidate
  Referenced from: /System/Library/Frameworks/ImageIO.framework/Versions/A/Resources/libPng.dylib
  Expected in: /Applications/gPodder.app/Contents/Resources/lib/libz.1.dylib
 in /System/Library/Frameworks/ImageIO.framework/Versions/A/Resources/libPng.dylib

Traceback (most recent call last):
  File "/Applications/gPodder.app/Contents/Resources/bin/gpodder", line 155, in <module>
    main()
  File "/Applications/gPodder.app/Contents/Resources/bin/gpodder", line 148, in main
    from gpodder.gtkui import main
  File "/Applications/gPodder.app/Contents/Resources/lib/python2.7/site-packages/gpodder/gtkui/main.py", line 22, in <module>
    import gtk
  File "/Applications/gPodder.app/Contents/Resources/lib/python2.7/site-packages/gtk-2.0/gtk/__init__.py", line 30, in <module>
    import gobject as _gobject
  File "/Applications/gPodder.app/Contents/Resources/lib/python2.7/site-packages/gobject/__init__.py", line 26, in <module>
    from glib import spawn_async, idle_add, timeout_add, timeout_add_seconds, \
  File "/Applications/gPodder.app/Contents/Resources/lib/python2.7/site-packages/glib/__init__.py", line 22, in <module>
    from glib._glib import *
ImportError: dlopen(/Applications/gPodder.app/Contents/Resources/lib/python2.7/site-packages/glib/_glib.so, 2): Symbol not found: _inflateValidate
  Referenced from: /System/Library/Frameworks/ImageIO.framework/Versions/A/Resources/libPng.dylib
  Expected in: /Applications/gPodder.app/Contents/Resources/lib/libz.1.dylib
 in /System/Library/Frameworks/ImageIO.framework/Versions/A/Resources/libPng.dylib
```

In console I get this:

Process: `syspolicyd`

Message #1: 
`temporarySigning type=1 matchFlags=0x0 path=/Applications/gPodder.app`

Message #2: 
`assessment granted for gPodder-3.9.3_0.zip by _XProtect
com.apple.message.domain: com.apple.security.assessment.outcome2
com.apple.message.signature2: bundle:UNBUNDLED
com.apple.message.signature3: gPodder-3.9.3_0.zip
com.apple.message.signature5: UNKNOWN
com.apple.message.signature4: 3
com.apple.message.signature: granted:_XProtect`

Message #3:
`assessment denied for .app
com.apple.message.domain: com.apple.security.assessment.outcome2
com.apple.message.signature2: bundle:org.gpodder
com.apple.message.signature3: .app
com.apple.message.signature5: 3.9.3
com.apple.message.signature4: 1
com.apple.message.signature: denied:no usable signature`

Message #4:
`added rule for .app
com.apple.message.domain: com.apple.security.assessment.update
com.apple.message.signature2: bundle:org.gpodder
com.apple.message.signature3: .app
com.apple.message.signature5: 3.9.3`

Message #5:
`com.apple.message.domain: com.apple.security.assessment.whitelist2
com.apple.message.signature: org.gpodder
com.apple.message.signature2: 76ac25c4c65320dea0077fc4ac01b479390486a6
com.apple.message.signature3: 40b46501114372a7744c44062633966a1c7265bb
com.apple.message.result: pass
com.apple.message.reason: -67015
com.apple.message.version: 3.9.3
com.apple.message.version2: 3.9.3
com.apple.message.execname: gpodder`

Message #6:
`assessment granted for .app by allowed cdhash
com.apple.message.domain: com.apple.security.assessment.outcome2
com.apple.message.signature2: bundle:org.gpodder
com.apple.message.signature3: .app
com.apple.message.signature5: 3.9.3
com.apple.message.signature4: 1
com.apple.message.signature: granted:allowed cdhash`

Message #7:
`OSActivityID: 0x80000000000317a1
com.apple.message.domain: com.apple.coreservices.app_capability_is_disabled.nap
com.apple.message.signature: org.gpodder
com.apple.message.signature2: 3.9.3
com.apple.message.value: 1`

Message #8:
`OSActivityID: 0x80000000000317a9
com.apple.message.domain: com.apple.coreservices.app_capability_is_disabled.nap
com.apple.message.signature: org.gpodder
com.apple.message.signature2: 3.9.3
com.apple.message.value: 0`

Thank you and hope this helps!

**Comment by elelay** [comment]
The stack trace from terminal really shows the problem: for some reason, the system's library `/System/Library/Frameworks/ImageIO.framework/Versions/A/Resources/libPng.dylib` gets picked instead of our bundled libpng. In turn, that system library has expectations on the  `libz.1.zlib` that we bundle, but in a too old version.
I'll try to untangle this and let you know.
Thanks for the great info!

**Comment by elelay** [comment]
Hi @jonathangodoy,
can you please try this pre-release: https://sourceforge.net/projects/gpodder/files/macosx/gPodder-3.9.3_1-alpha.zip/download
I've bumped all library versions, including zlib (1.2.8 -> 1.2.11 which has the `_inflateValidate` symbol).
Thanks,

**Comment by jonathangodoy** [comment]
Hi @elelay,

Yes, this pre-release version is working for me now.

Thanks for working on this, I appreciate it.

If you need me for further debugging, let me know.

**Comment by isimmons** [comment]
Hey just updated to High Sierra. Had the same problem but the pre-release works here too. Thanks

**Comment by elelay** [comment]
Good to know that it works, thanks @jonathangodoy!
I'll do a formal release soon (updated readme, commit changes), but you're good to go with that one.

**Comment by elelay** [comment]
Released 3.9.3_1
